src/main/java/org/jenkinsci/plugins/workflow/support/steps/ExecutorStepExecution.java

```diff
@@ -399,20 +404,25 @@ private static void cancelOwnerExecution(Node node, CauseOfInterruption... cause
         }
     }
 
-    public static final class RemovedNodeCause extends CauseOfInterruption {
+    public static final class RemovedNodeCause extends RetryableCauseOfInterruption {
         @SuppressFBWarnings(value = "MS_SHOULD_BE_FINAL", justification = "deliberately mutable")
         public static boolean ENABLED = Boolean.parseBoolean(System.getProperty(ExecutorStepExecution.class.getName() + ".REMOVED_NODE_DETECTION", "true"));
         @Override public String getShortDescription() {
             return "Agent was removed";
         }
     }
 
-    public static final class RemovedNodeTimeoutCause extends CauseOfInterruption {
+    public static final class RemovedNodeTimeoutCause extends RetryableCauseOfInterruption {
         @Override public String getShortDescription() {
             return "Timeout waiting for agent to come back";
         }
     }
 
+    /**
+     * Base class for a cause of interruption that can be retried via {@link AgentErrorCondition}.
+      */
+    private abstract static class RetryableCauseOfInterruption extends CauseOfInterruption implements Retryable {}
```

Comment:
Does not seem to contribute anything beyond extending those supertypes directly?

Reply:
Yes, this is just to avoid boilerplate. Can inline just as well.